hydromt/models/model_lumped.py

```diff
@@ -153,56 +183,67 @@ def __init__(
             data_libs=data_libs,
             logger=logger,
         )
-
-    def read(
+    
+    def setup_response_unit(
         self,
-        components: List = [
-            "config",
-            "response_units",
-            "geoms",
-            "forcing",
-            "states",
-            "results",
-        ],
-    ) -> None:
-        """Read the complete model schematization and configuration from model files.
-
-        Parameters
-        ----------
-        components : List, optional
-            List of model components to read, each should have an associated read_<component> method.
-            By default ['config', 'maps', 'response_units', 'geoms', 'forcing', 'states', 'results']
+        split_regions = False,
+        split_method = "us_area",
+        hydrography_fn = "merit_hydro",
+        **kwargs
+        ):
+        """
+        This component sets up the region and response units.
+        
         """
-        super().read(components=components)
 
-    def write(
+        if len(self.region) == 0:
+            raise ValueError("No region defined. Define :py:meth:`~hydromt.models.model_api.setup_region`")
+        
+        if not split_regions:
+            self.set_response_units(self._geoms['region'])
```

Comment:
I'm wondering if this will work as we don't yet support GeoDataFrames in set_response_units. Also I would always use `self.region` instead of `self._geoms['region']`.

Reply:
Good point. I started the original by just parsing a geodataframe with geometries, then changed to derive the region based on points. Good example why multiple tests are needed.

On topic: I think parsing a geoDataFrame into the response unit is a good functionality to have. For example when reading external (preprocessed) geoDataFrames Either standalone, or in the set_response_units function.

Reply:
This is now solved by adding the function gpd_to_response_units in workflows.to_response_units